scipy/stats/_continuous_distns.py

```diff
@@ -8703,6 +8703,15 @@ def _logsf(self, x, a, b):
             logsf[i] = np.log1p(-np.exp(self._logcdf(x[i], a[i], b[i])))
         return logsf
 
+    def _entropy(self, a, b):
+        A = 0.5 * (1 + sc.erf(a / np.sqrt(2)))
+        B = 0.5 * (1 + sc.erf(b / np.sqrt(2)))
+        Z = B - A
+        C = (np.log(np.pi) + np.log(2) + 1) / 2
+        D = np.log(Z)
```

Comment:
I do not understand how this is the same as the formula found for example in wikipedia:

![image](https://user-images.githubusercontent.com/40656107/215257202-48044039-40be-4b39-be57-7f4d13c85c59.png)
![image](https://user-images.githubusercontent.com/40656107/215257221-1aed1642-13dd-4ab2-b966-645605f7f2eb.png)

The second term using the PDF of the standard normal distribution is missing.

With $C=...$ I guess you are taking the first logarithmic term apart, this is a good idea

Please compare this with your closed PR where I had suggested a [formulation](https://github.com/scipy/scipy/pull/17774/commits/a6059537a51e328e643a2f48154ba0764c91c5b5) as many existing functions can be reused here. The standard normal CDF throughout this file is used as `_norm_cdf` and the standard normal PDF as `_norm_pdf` .

Reply:
>I do not understand how this is the same as the formula found for example in wikipedia

That's because it's not the same formula. The Wikipedia formula is wrong, so I derived the true formula by hand. I considered putting a comment explaining this, but I'm planning on changing the Wikipedia page once this PR is pulled, so it wouldn't make sense for me to do that.

Reply:
Please double-check before changing the Wikipedia entry. It's possible that the first three implementations here are incorrect, but I would be very surprised if that were the case.
```
import numpy as np
from scipy import stats
from scipy import special as sc
from mpmath import mp
mp.dps = 50

def entropy_wikipedia(a, b):
    Z = stats.norm.cdf(b) - stats.norm.cdf(a)
    return (np.log(np.sqrt(2 * np.pi * np.e) * Z)
            + (a * stats.norm.pdf(a) - b * stats.norm.pdf(b))/(2*Z))

def entropy_mpmath(a, b):
    Z = mp.ncdf(b) - mp.ncdf(a)
    def pdf(x):
        return mp.npdf(x)/Z
    def dentropy(x):
        y = pdf(x)
        return -mp.log(y) * y
    return mp.quad(dentropy, (a, b))

def entropy_pr(a, b):
    A = 0.5 * (1 + sc.erf(a / np.sqrt(2)))
    B = 0.5 * (1 + sc.erf(b / np.sqrt(2)))
    Z = B - A
    C = (np.log(np.pi) + np.log(2) + 1) / 2
    D = np.log(Z)
    h = (C + D) / (2 * Z)
    return h

a, b = -1, 3
print(entropy_wikipedia(a, b))  # 1.0926338726407065
print(entropy_mpmath(a, b))  # 1.0926338726407065848966922217503260384942743385556
print(stats.truncnorm.entropy(a, b))  # 1.0926338726407065
print(entropy_pr(a, b))  # 0.7408253846201
```

Reply:
> Please double-check before changing the Wikipedia entry. It's possible that the first three implementations here are incorrect, but I would be very surprised if that were the case.

You're right. I was using the formula for 0 to infinity, when I was actually supposed to be using the formula from a to b. Fixed it.